extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/DbCheckingPMetaSupplierBuilder.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.mongodb.impl;
+
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.core.Vertex;
+import com.hazelcast.jet.pipeline.DataConnectionRef;
+import com.mongodb.client.MongoClient;
+
+import java.security.Permission;
+
+public class DbCheckingPMetaSupplierBuilder {
+    private Permission requiredPermission;
+    private boolean checkResourceExistence;
+    private boolean forceTotalParallelismOne;
+    private String databaseName;
+    private String collectionName;
+    private SupplierEx<? extends MongoClient> clientSupplier;
+    private DataConnectionRef dataConnectionRef;
+    private ProcessorSupplier processorSupplier;
+    private int preferredLocalParallelism = Vertex.LOCAL_PARALLELISM_USE_DEFAULT;
+
+    public DbCheckingPMetaSupplierBuilder setRequiredPermission(Permission requiredPermission) {
+        this.requiredPermission = requiredPermission;
+        return this;
+    }
+
+    public DbCheckingPMetaSupplierBuilder setCheckResourceExistence(boolean checkResourceExistence) {
+        this.checkResourceExistence = checkResourceExistence;
+        return this;
+    }
+
+    public DbCheckingPMetaSupplierBuilder setForceTotalParallelismOne(boolean forceTotalParallelismOne) {
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        return this;
+    }
+
+    public DbCheckingPMetaSupplierBuilder setDatabaseName(String databaseName) {
```

Comment:
nit: builders usually do not prefix method names with `set`

Reply:
7831956e3b2ce0787afc566a54b89fa469b724b0